lib/diagnostic.js

```diff
@@ -36,7 +48,9 @@ const newArray = function newArray(length, defaultValue) {
 // The function you write should return
 // the object passed in as an argument or the object created
 const addMethod = function addMethod(propertyName, method, obj) {
-
+  let returnedObj = obj || new Object();
+  returnedObj.propertyName = method;
```

Comment:
Good thought, but not quite right. This will literally set a property called `propertyName`, not whatever String you would have passed in as an argument.